diff --git a/utils/native/lite/hals/file/hal_file.h b/utils/native/lite/hals/file/hal_file.h
index bfcf3a293..bee85378c 100755
--- a/utils/native/lite/hals/file/hal_file.h
+++ b/utils/native/lite/hals/file/hal_file.h
@@ -36,6 +36,13 @@ int HalFileStat(const char* path, unsigned int* fileSize);
 
 int HalFileSeek(int fd, int offset, unsigned int whence);
 
+int FileRead(const char *filename,	unsigned int offset, char *buf,  unsigned int len);
+
+int FileWrite(const char *filename,  unsigned int offset, const char *buf,	unsigned int len);
+
+int FileSize(const char *filename);
+
+
 #ifdef __cplusplus
 #if __cplusplus
 }
