diff --git a/utils/native/lite/hals/file/hal_file.h b/utils/native/lite/hals/file/hal_file.h
index bfcf3a29..bc5a94f7 100755
--- a/utils/native/lite/hals/file/hal_file.h
+++ b/utils/native/lite/hals/file/hal_file.h
@@ -36,10 +36,17 @@ int HalFileStat(const char* path, unsigned int* fileSize);
 
 int HalFileSeek(int fd, int offset, unsigned int whence);
 
+int FileRead(const char *filename,	unsigned int offset, char *buf,  unsigned int len);
+
+int FileWrite(const char *filename,  unsigned int offset, const char *buf,	unsigned int len);
+
+int FileSize(const char *filename);
+
+
 #ifdef __cplusplus
 #if __cplusplus
 }
 #endif
 #endif /* __cplusplus */
 
-#endif  // HAL_FILE_SYSTEM_API_H
\ No newline at end of file
+#endif  // HAL_FILE_SYSTEM_API_H
